portal/models/app_text.py

```diff
@@ -416,3 +419,14 @@ def app_text(name, *args):
         raise ValueError(
             "AppText with name '{}' defines more parameters "
             "than provided: `{}`".format(name, *args))
+
+
+def format_asset_attributes(asset, variables):
+    asset_out = ""
+    x = 0
+    for match in finditer(r"\${[^}]+}", asset):
```

Comment:
I wasn't sure if this would work, because LR is sending the variables in `${...}` format, whereas string format just looks for `{...}`, right? I guess we could just use string format and then remove all $'s, but that runs the risk of removing $'s unrelated to the variables (edge case, but it could happen)